### PR TITLE
[SIM_FLUTTER-62][FE] Create account details page

### PR DESCRIPTION
**Pull request by rdelrosariojr**
### Issue Links
- [Backlog](https://framgiaph.backlog.com/view/SIM_FLUTTER-62)

### Definition of Done
- [x] Create account details page

### Notes
- added button to homepage in order to navigate to the account details page

### Attachments

https://github.com/framgia/sph-flutter/assets/97145763/f13ce140-50ab-4c3c-9ad6-64c3a365f60f

### CODE REVIEW

**Comment by Dbulawan13** [comment]
Comment:

- [ ] When user click the percentage of the donut graph. It should expand based on figma.
For Reference: https://pub.dev/packages/fl_chart

Once this PR is merge https://github.com/framgia/sph-flutter/pull/12 and https://github.com/framgia/sph-flutter/pull/16 . Please update this so that the buttons will work properly.
Expected result:

https://github.com/framgia/sph-flutter/assets/86758474/42b4e319-0adc-441c-a4f9-4e33105eaca8

Comment: 1

**Comment by Dbulawan13** [comment]
Comment:

- [ ] Deposit, Withdraw, and Transfer component will not appear once user click on the button

https://github.com/framgia/sph-flutter/assets/86758474/0b20bea0-f015-48c5-baa2-43e1680a94f8

Comment: 1

File: frontend/lib/src/components/balance_card.dart [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import 'package:flutter/material.dart';
+
+class BalanceCard extends StatelessWidget {
```
Comment (jstephend-sun) [inline frontend/lib/src/components/balance_card.dart:9]: provide description for the widget and its parameters

File: frontend/lib/src/components/button.dart [hunk 2]
```diff
@@ -15,31 +15,43 @@ class Button extends StatelessWidget {
     required this.text,
     this.onPressed,
     this.padding,
+    this.buttonColor = const Color(0xFF00CCFF),
```
Comment (jstephend-sun) [inline frontend/lib/src/components/button.dart:26]: add descriptions for the parameters

File: frontend/lib/src/components/graph.dart [hunk 3]
```diff
@@ -0,0 +1,178 @@
+import 'package:flutter/material.dart';
+import 'package:fl_chart/fl_chart.dart';
+
+import 'package:frontend/src/components/indicator.dart';
+
+class Graph extends StatefulWidget {
```
Comment (jstephend-sun) [inline frontend/lib/src/components/graph.dart:13]: use of stateless widget

Comment (jstephend-sun) [inline frontend/lib/src/components/graph.dart:13]: provide description to the widget

File: frontend/lib/src/components/graph.dart [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+  const Graph({super.key});
+
+  @override
+  State<StatefulWidget> createState() => GraphState();
+}
+
+class GraphState extends State {
+  int touchedIndex = -1;
```
Comment (jstephend-sun) [inline frontend/lib/src/components/graph.dart]: use getxController

**Review by Dbulawan13 — DISMISSED** [review]
Approve PR #14 

Working:
- I am able to create an account details page.
- I am able to display the Deposit, Withdraw, and Transfer components once the button is clicked.
- I will redirect to the transaction history page once the transaction button is clicked.

https://github.com/framgia/sph-flutter/assets/86758474/35f9fb2b-cf1d-4a83-b209-ba90ee9302b1

Testing Hours: 2
Comments: 2
Iterations: 2

File: frontend/lib/src/components/graph.dart (hunk 4 above)
Comment (jstephend-sun) [inline frontend/lib/src/components/graph.dart]: can we use other name like focus? haha

File: frontend/lib/src/components/indicator.dart [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import 'package:flutter/material.dart';
+
+class Indicator extends StatelessWidget {
```
Comment (jstephend-sun) [inline frontend/lib/src/components/indicator.dart:17]: widget description

File: frontend/lib/src/components/indicator.dart [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  const Indicator({
+    super.key,
+    required this.color,
+    required this.text,
+    required this.isSquare,
+    this.size = 16,
+    this.textColor,
+  });
+  final Color color;
```
Comment (jstephend-sun) [inline frontend/lib/src/components/indicator.dart]:
```suggestion
  });
  
  final Color color;
```
space

File: frontend/lib/src/components/indicator.dart [hunk 7]
```diff
@@ -0,0 +1,44 @@
+import 'package:flutter/material.dart';
+
+class Indicator extends StatelessWidget {
+  const Indicator({
+    super.key,
+    required this.color,
```
Comment (jstephend-sun) [inline frontend/lib/src/components/indicator.dart:20]: parameter descriptions

File: frontend/lib/src/components/indicator.dart [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+            shape: isSquare ? BoxShape.rectangle : BoxShape.circle,
+            color: color,
+          ),
+        ),
+        const SizedBox(
+          width: 4,
+        ),
+        Text(
+          text,
+          style: TextStyle(
```
Comment (jstephend-sun) [inline frontend/lib/src/components/indicator.dart]: use label large and apply bold

File: frontend/lib/src/features/account_details/account_details_page.dart [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+
+class AccountDetailsPage extends StatelessWidget {
+  const AccountDetailsPage({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final arguments =
+        ModalRoute.of(context)!.settings.arguments as ScreenArguments;
+
+    // this is temporary, this is to verify that the argument was pass successfully
```
Comment (jstephend-sun) [inline frontend/lib/src/features/account_details/account_details_page.dart]: add TODO in the comment so it can be found easily

**Review by RDLegs — DISMISSED** [review]
Looks good to me!

You missed to add accountId in `toJson()` but we can change that during the implementation task

File: frontend/lib/src/models/account.dart [hunk 10]
```diff
@@ -9,17 +9,20 @@ Account accountFromJson(String str) => Account.fromJson(json.decode(str));
...
 
   Account({
+    required this.accountId,
     required this.userId,
     required this.accountType,
     required this.name,
   });
 
   factory Account.fromJson(Map<String, dynamic> json) => Account(
+        accountId: json["account_id"],
```
Comment (jstephend-sun) [inline frontend/lib/src/models/account.dart]: wrong key, refer to database

**Review by jstephend-sun — CHANGES_REQUESTED** [review]
Requested Changes #14

review time: 10mins
comments: 10

File: frontend/lib/src/components/graph.dart (hunk 3 above)
Comment (rdelrosariojr) [reply frontend/lib/src/components/graph.dart:13]: @jstephend-sun I declared dynamic variable focusIndex, correct me if I'm wrong, we can't use stateless for this one.

**Review by jstephend-sun — APPROVED** [review]
Approved #14

review time: 10mins
comments: 10

**Pull request merged**